ui_framework/services/sort/sort_properties.test.js

```diff
@@ -0,0 +1,137 @@
+import _ from 'lodash';
+import {
+  SortProperties,
+} from './sort_properties';
+
+describe('SortProperties', () => {
+  const name = {
+    name: 'name',
+    getValue: bird => bird.name,
+    isAscending: true,
+  };
+
+  const size = {
+    name: 'size',
+    getValue: bird => bird.size,
+    isAscending: false,
+  };
+
+  const color = {
+    name: 'color',
+    getValue: bird => bird.color,
+    isAscending: true,
+  };
+
+  const birds = [
+    {
+      name: 'cardinal',
+      color: 'red',
+      size: 7,
+    },
+    {
+      name: 'bluejay',
+      color: 'blue',
+      size: 8,
+    },
+    {
+      name: 'chickadee',
+      color: 'black and white',
+      size: 3,
+    }
+  ];
+
+  describe('initialSortProperty', () => {
+    test('is set', () => {
+      const sortProperties = new SortProperties([name, size, color], 'size');
+      expect(sortProperties.getSortProperty().name).toBe('size');
+    });
+
+    test('throws an error with an invalid property name', () => {
+      expect(new SortProperties([name, size, color], 'does not exist')).toThrow();
+    });
+
+    test('is optional', () => {
```

Comment:
Maybe this shouldn't be optional. Should we throw an error if it isn't supplied? I think we always want to set a default sort for anything that can be sorted.